### PR TITLE
minor: port date_bin tests to sqllogictests

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/4853

# Rationale for this change

I plan to add some additional coercion for date_bin and I want to write the tests using sqllogictest. This PR moves the test. I will build on this PR for https://github.com/apache/arrow-datafusion/issues/4853

# What changes are included in this PR?

port `date_bin` test to use sqllogictest framework 

# Are these changes tested?
yes (they are tests)

# Are there any user-facing changes?
no

### CODE REVIEW

**Comment by alamb** [comment]
FYI @stuartcarnie as the original author of these tests

**Comment by stuartcarnie** [comment]
Very nice – looks like the test files are YAML?

**Comment by alamb** [comment]
> 
Very nice – looks like the test files are YAML?



The test files are actually sqllite style "sqllogictest" files -- details on the format an be found in https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sqllogictests/README.md

**Comment by alamb** [comment]
@xudong963  or @melgenek  could I trouble one of you for a review of this PR?

File: datafusion/core/tests/sqllogictests/test_files/timestamps.slt [hunk 1]
```diff
@@ -67,3 +67,69 @@ select * from foo where ts != '2000-02-01T00:00:00';
...
+
+# origin shifts bins forward 1 day
+query T
+SELECT DATE_BIN(INTERVAL '7 days', TIMESTAMP '2022-08-03 14:38:50Z', TIMESTAMP '1970-01-02 00:00:00Z') AS res
+----
+2022-07-29T00:00:00
+
+
+# demonstrates array values (rather than scalar) for the source argument
+query T
```
Comment (xudong963) [inline datafusion/core/tests/sqllogictests/test_files/timestamps.slt:113]:
```suggestion
query T rowsort
```

**Pull request merged**